src/TezosProvider.ts

```diff
@@ -0,0 +1,509 @@
+import axios from "axios";
+
+import { UniversalProvider, Metadata } from "@walletconnect/universal-provider";
+import { KeyValueStorageOptions } from "@walletconnect/keyvaluestorage";
+import { Logger } from "@walletconnect/logger";
+import { TezosToolkit } from "@taquito/taquito";
+import {
+  PartialTezosDalPublishCommitmentOperation,
+  PartialTezosDelegationOperation,
+  PartialTezosIncreasePaidStorageOperation,
+  PartialTezosOperation as PartialTezosOperationOriginal,
+  PartialTezosOriginationOperation as PartialTezosOriginationOperationOriginal,
+  PartialTezosRegisterGlobalConstantOperation,
+  PartialTezosRevealOperation,
+  PartialTezosSetDepositsLimitOperation,
+  PartialTezosSmartRollupAddMessagesOperation,
+  PartialTezosSmartRollupCementOperation,
+  PartialTezosSmartRollupExecuteOutboxMessageOperation,
+  PartialTezosSmartRollupOriginateOperation,
+  PartialTezosSmartRollupPublishOperation,
+  PartialTezosSmartRollupRecoverBondOperation,
+  PartialTezosSmartRollupRefuteOperation,
+  PartialTezosSmartRollupTimeoutOperation,
+  PartialTezosTransactionOperation,
+  PartialTezosTransferTicketOperation,
+  PartialTezosUpdateConsensusKeyOperation,
+  TezosActivateAccountOperation,
+  TezosBallotOperation,
+  TezosFailingNoopOperation,
+  TezosOperationType,
+  TezosProposalOperation,
+} from "@airgap/beacon-types";
+
+import { ScriptedContracts } from "@taquito/rpc";
+import {
+  AssetData,
+  ChainsMap,
+  TezosConnectionError,
+  TezosConnectOpts,
+  TezosGetAccountResponse,
+  TezosInitializationError,
+  TezosMethod,
+  TezosProviderError,
+  TezosSendResponse,
+  TezosSignResponse,
+} from "./types";
+import {
+  DefaultTezosMethods,
+  RelayUrl,
+  TezosChainDataMainnet,
+  TezosChainDataTestnet,
+  TezosChainMap,
+  UnsupportedOperations,
+} from "./constants";
+import { SessionTypes } from "@walletconnect/types";
+
+interface PartialTezosOriginationOperation
+  extends Omit<PartialTezosOriginationOperationOriginal, "script"> {
+  script: ScriptedContracts;
+}
+
+type PartialTezosOperation =
+  | Exclude<PartialTezosOperationOriginal, PartialTezosOriginationOperationOriginal>
+  | PartialTezosOriginationOperation;
+
+export interface TezosProviderOpts {
+  projectId: string;
+  metadata: Metadata;
+  relayUrl?: string;
+  storageOptions?: KeyValueStorageOptions;
+  disableProviderPing?: boolean;
+  logger?: string | Logger; // default: "info"
+}
+
+// Provides a way to interact with the Tezos blockchain.
+// Secures that WalletConnect is used with PartialTezosOperation
+export class TezosProvider {
+  public namespace: string = "tezos";
+  public signer?: InstanceType<typeof UniversalProvider> = undefined;
```

Comment:
at a quick glance, it seems that TezosProvider is a broken object without signer, tezosToolkit, etc. defined. why not to make those non-optional, require them in the constructor and make the init static function prepare all those arguments *in advance*?